google/cloud/storage/bucket_metadata.h

```diff
@@ -15,14 +15,74 @@
 #ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_BUCKET_METADATA_H_
 #define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_BUCKET_METADATA_H_
 
+#include "google/cloud/internal/optional.h"
+#include "google/cloud/storage/bucket_access_control.h"
 #include "google/cloud/storage/internal/common_metadata.h"
 #include "google/cloud/storage/internal/nljson.h"
+#include "google/cloud/storage/object_access_control.h"
 #include <map>
+#include <tuple>
+#include <utility>
 
 namespace google {
 namespace cloud {
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
+/**
+ * The billing configuration for a Bucket.
+ *
+ * @see https://cloud.google.com/storage/docs/requester-pays for general
+ *     information on "Requester Pays" billing.
+ */
+struct BucketBilling {
+  BucketBilling() : requester_pays(false) {}
+
+  bool requester_pays;
+};
+
+/**
+ * An entry in the CORS list.
+ *
+ * CORS (Cross-Origin Resource Sharing) is a mechanism to enable client-side
+ * cross-origin requests. An entry in the configuration has a maximum age and a
+ * list of allowed origin and methods, as well as a list of returned response
+ * headers.
+ *
+ * @see https://en.wikipedia.org/wiki/Cross-origin_resource_sharing for general
+ *     information on CORS.
+ *
+ * @see https://cloud.google.com/storage/docs/cross-origin for general
+ *     information about CORS in the context of Google Cloud Storage.
+ *
+ * @see https://cloud.google.com/storage/docs/configuring-cors for information
+ *     on how to set and troubleshoot CORS settings.
+ */
+struct CorsEntry {
+  google::cloud::internal::optional<std::int64_t> max_age_seconds;
```

Comment:
[no action needed]

Adding the extra functionality to the `optional` class looks like it was... *fun* :P But having having `optional` here looks much more intuitive than trying use something like -1 to indicate absence of this field.  Thanks!